src/modules/m_nationalutf.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2014 Googolplexed <googol@googolplexed.net>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "inspircd.h"
+#include "caller.h"
+
+class ModuleNationalUTF : public Module,public HandlerBase1<bool, const std::string&>
+{
+	caller1<bool, const std::string&> rememberer;
+	struct Range {
+		unsigned long min;
+		unsigned long max;
+		Range(unsigned long mi,unsigned long ma)
+		{
+			min = mi;
+			max = ma;
+		};
+	};
+	std::vector<Range> ranges;
+
+	//Returns the next character in UTF32 encoding and advances the iterator
+	//past it. Returns 0 on failure
+	unsigned long utf8to32(std::string str,std::string::const_iterator& it)
+	{
+
+		/*
+		 * UTF8 is a variable length encoding
+		 * The first UTF byte determines the number of following bytes
+		 * We set aside four 8-bit bitsets to hold the theoretical maximum number of bytes
+		 * As the initial header byte uses a different header, we keep that aside
+		 * as 'ltr'
+		 */
+		std::bitset<8> ltr = *it;
+		std::bitset<8> sets[3];
+		/*
+		 * As per the UTF spec, the number of initial true bits
+		 * determines the number of following bytes.
+		 * For each true bit, we advance the iterator forward and copy
+		 * the byte into the the relevant bitset.
+		 * We also clean up the initial headers of the first byte so we
+		 * don't have to do this later.
+		 */
+
+		//MSB is 7, but we've already handled that in ltr, so start at 6
+		//Only a maximum of 4 bits can be set, including the initial
+		int bytes = 1; //Already got a header byte
+		for(int i=6; ltr[i]!=0 && i>3; i--)
+		{
+			//For true bit in the UTF header, we grab an extra byte.
+			it++;
+			if(it == str.end()) return 0;         //Prevent invalid header from iterating too far.
```

Comment:
These should be one separate lines, and the comment should be either on its own line or next to the `if` separated by one space. C++-style single line comments should have a space between the `//` and the text.

    if (it == str.end()) // Prevent invalid header from iterating too far.
	    return 0;

or

    // Prevent invalid header from iterating too far.
    if (it == str.end())
	    return 0;